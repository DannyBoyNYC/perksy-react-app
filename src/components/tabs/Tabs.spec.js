const React = require('react');

const Tabs = require('./index').default;

describe('<Tabs.* />', () => {
  let component;
  const requiredProps = {
    children: [
      <Tabs.Panel title="Title for content 1">
        <p>Content 1</p>
      </Tabs.Panel>,
      <Tabs.Panel title="Title for content 2">
        <p>Content 2</p>
      </Tabs.Panel>
    ]
  };

  describe('<Tabs.Horizontal />', () => {
    beforeEach(() => {
      component = mountWithTheme(
        <Tabs.Horizontal {...requiredProps} />,
        'Tabs'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render each Panel', () => {
      expect(component).toContainMatchingElements(2, 'Panel');
    });

    it('should default the active Panel to the first Panel', () => {
      component = shallowUntilTarget(
        <Tabs.Horizontal {...requiredProps} />,
        'Tabs'
      );
      expect(component).toHaveState({ activeTabIndex: 0 });
    });

    it('should render the content of the active Panel', () => {
      expect(component).toIncludeText('Content 1');
    });

    it('should update the active Panel on click', () => {
      component
        .find('Panel')
        .last()
        .simulate('click');
      expect(component).toIncludeText('Content 2');
    });
  });
});
