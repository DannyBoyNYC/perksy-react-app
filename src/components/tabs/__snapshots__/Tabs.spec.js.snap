// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs.* /> <Tabs.Horizontal /> should render correctly with all required props 1`] = `
<Horizontal>
  <Jss(Tabs)
    variation="horizontal"
  >
    <Tabs
      classes={
        Object {
          "content": "Tabs-content-0-1-2",
          "horizontal": "Tabs-horizontal-0-1-5",
          "root": "Tabs-root-0-1-1",
          "tabItem": "Tabs-tabItem-0-1-3",
          "tabList": "Tabs-tabList-0-1-4",
        }
      }
      theme={
        Object {
          "background": Object {
            "block": "#F7F6F5",
            "item": "#F7F6F5",
            "main": "#FFFFFF",
            "page": "#FFFFFF",
            "sidebar": "#19222F",
          },
          "colors": Object {
            "base": "#BDBDBD",
            "black": "#000000",
            "dark": "#5F5E66",
            "invalid": "red",
            "light": "#DEDDDC",
            "lightest": "#F7F6F5",
            "primary": "#4797F4",
            "primaryDark": "#2E7EDB",
            "primaryLight": "#61B1FF",
            "secondary": "#19222F",
            "valid": "#58C087",
            "white": "#FFFFFF",
          },
          "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": Object {
            "base": "15px",
            "h1": "28px",
            "h2": "22px",
            "h3": "20px",
            "h4": "18px",
            "h5": "18px",
            "h6": "16px",
            "large": "20px",
            "small": "13px",
            "text": "16px",
          },
        }
      }
      variation="horizontal"
    >
      <div
        className="Tabs-root-0-1-1 Tabs-horizontal-0-1-5"
      >
        <ul
          className="Tabs-tabList-0-1-4"
          role="tablist"
        >
          <Jss(Panel)
            isActive={true}
            key=".0"
            onClick={[Function]}
            tabIndex={0}
            title="Title for content 1"
          >
            <Panel
              classes={
                Object {
                  "content": "Panel-content-0-1-7",
                  "horizontal": "Panel-horizontal-0-1-10",
                  "root": "Panel-root-0-1-6",
                  "tabItem": "Panel-tabItem-0-1-8",
                  "tabList": "Panel-tabList-0-1-9",
                }
              }
              isActive={true}
              onClick={[Function]}
              tabIndex={0}
              theme={
                Object {
                  "background": Object {
                    "block": "#F7F6F5",
                    "item": "#F7F6F5",
                    "main": "#FFFFFF",
                    "page": "#FFFFFF",
                    "sidebar": "#19222F",
                  },
                  "colors": Object {
                    "base": "#BDBDBD",
                    "black": "#000000",
                    "dark": "#5F5E66",
                    "invalid": "red",
                    "light": "#DEDDDC",
                    "lightest": "#F7F6F5",
                    "primary": "#4797F4",
                    "primaryDark": "#2E7EDB",
                    "primaryLight": "#61B1FF",
                    "secondary": "#19222F",
                    "valid": "#58C087",
                    "white": "#FFFFFF",
                  },
                  "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": Object {
                    "base": "15px",
                    "h1": "28px",
                    "h2": "22px",
                    "h3": "20px",
                    "h4": "18px",
                    "h5": "18px",
                    "h6": "16px",
                    "large": "20px",
                    "small": "13px",
                    "text": "16px",
                  },
                }
              }
              title="Title for content 1"
            >
              <li
                className="Panel-tabItem-0-1-8 active"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="tab"
              >
                <span>
                  Title for content 1
                </span>
              </li>
            </Panel>
          </Jss(Panel)>
          <Jss(Panel)
            isActive={false}
            key=".1"
            onClick={[Function]}
            tabIndex={1}
            title="Title for content 2"
          >
            <Panel
              classes={
                Object {
                  "content": "Panel-content-0-1-7",
                  "horizontal": "Panel-horizontal-0-1-10",
                  "root": "Panel-root-0-1-6",
                  "tabItem": "Panel-tabItem-0-1-8",
                  "tabList": "Panel-tabList-0-1-9",
                }
              }
              isActive={false}
              onClick={[Function]}
              tabIndex={1}
              theme={
                Object {
                  "background": Object {
                    "block": "#F7F6F5",
                    "item": "#F7F6F5",
                    "main": "#FFFFFF",
                    "page": "#FFFFFF",
                    "sidebar": "#19222F",
                  },
                  "colors": Object {
                    "base": "#BDBDBD",
                    "black": "#000000",
                    "dark": "#5F5E66",
                    "invalid": "red",
                    "light": "#DEDDDC",
                    "lightest": "#F7F6F5",
                    "primary": "#4797F4",
                    "primaryDark": "#2E7EDB",
                    "primaryLight": "#61B1FF",
                    "secondary": "#19222F",
                    "valid": "#58C087",
                    "white": "#FFFFFF",
                  },
                  "fontFamily": "Larsseit, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontFamilyBold": "Larsseit_Bold, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontFamilyMed": "Larsseit_Medium, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": Object {
                    "base": "15px",
                    "h1": "28px",
                    "h2": "22px",
                    "h3": "20px",
                    "h4": "18px",
                    "h5": "18px",
                    "h6": "16px",
                    "large": "20px",
                    "small": "13px",
                    "text": "16px",
                  },
                }
              }
              title="Title for content 2"
            >
              <li
                className="Panel-tabItem-0-1-8"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="tab"
              >
                <span>
                  Title for content 2
                </span>
              </li>
            </Panel>
          </Jss(Panel)>
        </ul>
        <div
          className="Tabs-content-0-1-2"
        >
          <p>
            Content 1
          </p>
        </div>
      </div>
    </Tabs>
  </Jss(Tabs)>
</Horizontal>
`;
